File: Recursion/Basic/C++/Print N natual Numbers in Reverse Order.cpp
#include <iostream>
using namespace std;

void printNaturalNumbers(int n) {
    if (n <= 0) // BASE CONDITION.
        return;
    
    cout << n << " ";
    printNaturalNumbers(n - 1);  // Recursive call.
    
}

int main() {
    int n;
    cout << "Enter value of n: ";
    cin >> n;

    cout << "First " << n << " natural numbers: ";
    printNaturalNumbers(n);
    cout << endl;

    return 0;
}

File: Recursion/Basic/python/Print N natual Numbers in Reverse Order.py
def print_natural_numbers(n):
    if n <= 0:
        return
    
    print(n, end=" ")
    print_natural_numbers(n - 1)

n = int(input("Enter value of n: "))
print(f"First {n} natural numbers: ", end="")
print_natural_numbers(n)
print()